classes/TableMigrationCodeGenerator.php

```diff
@@ -490,12 +484,21 @@ protected function generateDefaultMethodCall($default, $column)
 
         $type = MigrationColumnType::toMigrationMethodName($typeName, $columnName);
 
-        if (in_array($type, MigrationColumnType::getIntegerTypes()) || 
+        if (in_array($type, MigrationColumnType::getIntegerTypes()) ||
             in_array($type, MigrationColumnType::getDecimalTypes()) ||
             $type == MigrationColumnType::TYPE_BOOLEAN) {
             return sprintf('->default(%s)', $default);
         }
 
+        // Allow `null` value for nullable columns
+        if ($column->getNotnull() === false) {
+            if (strtolower($default) === 'null' && $column->getNotnull() === false) {
```

Comment:
@bennothommo minor note, could be on one micro-microsecond faster: `$column->getNotnull() === false && strtolower($default) === 'null'` 😄

Reply:
@w20k I just realised the `getNotnull` call is repeated - I've removed that.